### PR TITLE
Upgrade supported Spark version to v2.0.2

### PR DESCRIPTION
**Pull request by yueawang**
Upgrade supported Spark version to v2.0.2

## How was this patch tested?

mvn test pass

### CODE REVIEW

**Comment by LuciferYang** [comment]
shall we supported Spark version to v2.1.1?

**Comment by yueawang** [comment]
Yes, will eventually upgrade to support start-of-the-art spark.

File: src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala [hunk 1]
```diff
@@ -29,9 +29,7 @@ import scala.concurrent.duration._
 import scala.language.existentials
 import scala.language.postfixOps
 import scala.util.control.NonFatal
-
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala]: keep this blank line

File: src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala [hunk 2]
```diff
@@ -29,9 +29,7 @@ import scala.concurrent.duration._
...
 import org.apache.commons.lang3.SerializationUtils
-
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala]: keep this also

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala [hunk 3]
```diff
@@ -196,7 +198,7 @@ case class DropIndex(
     allowNotExists: Boolean,
     partitionSpec: Option[TablePartitionSpec]) extends RunnableCommand {
 
-  override def children: Seq[LogicalPlan] = Seq(relation)
+  override lazy val containsChild: Set[TreeNode[_]] = Seq(relation).toSet
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: Set[QueryPlan[_]]

Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: maybe innerChildren is better?

Comment (yueawang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: innerChildren is not be resolved during resolve process, so I chose containsChild

Comment (yueawang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/indexPlans.scala]: containsChild is a member of TreeNode which has TreeNode as its template param, I can not override it to QueryPlan. Is there a way to do that??

**Pull request merged**